level1/expressions-types-values.md

```diff
@@ -100,25 +100,205 @@ Here are three general problem solving strategies:
+
+> Can you write it using `.` and `()` instead of space?
+
+*They might need some help here.  Once they've finished, reiterate the definition of method call syntax*
+
 
+> Is `"hello".take(3)` the same as `"hello" take 3`?  Are these the same as `"hello".take(1 + 2)`?
+
+*`"hello".take(3)` and `"hello" take 3` are different notation for the same expression*
+*`"hello".take(1 + 2) is a different expression.  We're trying to get them to understand the difference between notation of expressions and expressions themselves*
+
+We can call `1+2` using method syntax
+
+```
+1.+(2)
+```
+
+> Write `2 / 4` using method syntax
+
+*This is to confirm their understanding*
+
+> Write 1 + 2 + 3 using method syntax
+
+*This is to confirm their understanding of method chaining*
+
+> Can you write `hello.take(3).take(2)` in infix notation?
+
+*This is to confirm their understanding of method chaining*
+
+> Why would you use one notation over the other?
+
+*If this is a group, try and get them to discuss which style they prefer*
+
+Programs are meant to be read by people, as well as evaluated by computers.  `1 + 2` tends to be easier to read than `1.+(2)`
+
+## Types
+
+Types stop us from calling methods that don't exist.
+
+The type of an expression tells Scala what methods can exist, and what they evaluate to.
+
+Here are some examples:
+
+```
+"Bronte" / "Austen"
+```
+
+> What is the type of this expression?  What do you think this expression will evaluate to?
+
+*The expression doesn't have a type. By the same token, it can't be evaluated*
+
+Consider
+
+```
+"Bronte".take("Austen")
+```
+
+>  What do you think this expression should evaluate to?
+
+*This expression also can't be evaluated*
+
+Methods arguments also have types.
+
+Consider
+
+```
+"Bronte".take(1 + 2)
+```
+
+> Is it possible to evaluate this expression?
+
+*A difficult question, as we haven't explained that arguments can be expressions*
+
+Method arguments are also expressions.  This means they can take in expressions such as `1 + 2`, as well as literal expressions such as `3`.
+
+```
+"Bronte".take(1 / 0)
+```
+
+> Is it possible to evaluate this expression?
+
+*Yes (hopefully, they'll try to do so).  The result will be an error*
+
+> What is the difference between the error in `"Bronte".take("Austen")` and `"Bronte".take(1 / 0)`?
+
+*One is an error that prevents evaluation from happening, while the other is an error that happens during evaluation.  Try to avoid references to compile time and run time at this point.*
 
 ## Exercises
 
 See Creative Scala Chapter 2 for additional exercises
 
+1. What is an expression? What is a value?  What is a type?
 
-1. What is an expression? What is a value? What is an object?
+1. Identify the values, objects and expressions in the following code.
+
+```
+"hello".take(1 + 2).toUpperCase
+```
+
+Which parts use method call syntax?  Which use infix notation?
+
+Can you write this completely in method call syntax?
```

Comment:
Maybe we should just call it "method call notation" throughout, and avoid the term "syntax"?

Reply:
We do need to introduce the term at some point, as students will encounter it in other contexts. I think Creative Scala has more detail on this, but I'm not 100% certain.

Reply:
Creative Scala doesn't ever define the term *syntax*, and starts using it when describing method call syntax.  I do think it's important to explore notation at this point, especially for those with no prior experience, as they cannot distinguish between the textual representation of an expression and the expression itself.  Using different notation for the same expression nudges them towards this.

Perhaps syntax is something we should give a concrete definition to?

Reply:
Yes, that sounds sensible to me. I've opened an issue on Creative Scala to define "syntax" in there.